### PR TITLE
EZP-30898: Dropped CleanInstaller and cleaned up installer code

### PR DESCRIPTION
**Pull request by alongosz**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30898](https://jira.ez.no/browse/EZP-30898) blocking [EZP-30624](https://jira.ez.no/browse/EZP-30624)
| **Is required by** | #2752
| **Requires** | ezsystems/BehatBundle#100
| **Bug/Improvement**| yes
| **New feature**    | yes
| **Target version** | `master (8.0@dev)` for eZ Platform `v3.0`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | ezsystems/developer-documentation#759

This PR drops deprecated since eZ Platform `v2.5` `CleanInstaller`. ~Seems like a clean forward compatibility layer for those changes is not possible due to optional dependency on DoctrineSchema Bundle. Therefore there are required changes for eZ Platform 3.0 for any 3rd party package or project implementing custom installation process.~ Provided forward compatibility for CoreInstaller via #2756, which allows applying the required changes to v2.5 installation, so the upgrade process is smoother.

### Summary of changes

1. Dropped Service Container parameters:
    - [x] `ezplatform.installer.clean_installer.class`,
    - [x] `ezplatform.installer.db_based_installer.class`,
    - [x] `ezplatform.installer.install_command.class`.

2. Dropped Services:
    - [x] `ezplatform.installer.clean_installer` along with its implementation (`EzSystems\PlatformInstallerBundle\Installer\CleanInstaller`),

3. Renamed Services:
    - [x] `ezplatform.installer.db_based_installer` into `EzSystems\PlatformInstallerBundle\Installer\DbBasedInstaller`,
    - [x] `ezplatform.installer.install_command` into 
    - [x] `EzSystems\PlatformInstallerBundle\Command\InstallPlatformCommand`

4. Misc changes:
    - [x] Marked `InstallPlatformCommand` as `final`,
    - [x] Changed `InstallPlatformCommand` Service definition in DIC to use named arguments, adjusted Compiler pass accordingly,
    - [x] Documented changes.

### QA

Note: might be tested along with #2752 for [EZP-30624](https://jira.ez.no/browse/EZP-30624).

- [x] Sanity checks on the installation process for eZ Platform OSS, eZ Platform OSS Demo, eZ Platform EE, eZ Platform EE Demo.

**TODO**:
- [x] **Rebase after merging #2756**
- [x] Implement tests (seems like there's zero coverage for those, so I need to start from a scratch).
- [x] Investigate, if time allows, if we can improve something for v2.5 in terms of forward compatibility, so the migration is easier (#2756).
- [x] Align `ezplatform-ee-installer` with the changes (ezsystems/ezplatform-ee-installer#58).
- [x] Align `ezplatform-ee-demo` with the changes (ezsystems/ezplatform-ee-demo#155).
- [x] Align `ezplatform-demo` with the changes (ezsystems/ezplatform-demo#113).
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: doc/bc/changes-8.0.md [hunk 1]
```diff
@@ -55,6 +55,22 @@ Changes affecting version compatibility with former or future versions.
...
+  See eZ Platform Documentation for more details.
+
+* The `ezplatform.installer.db_based_installer.class` and `ezplatform.installer.install_command.class`
+  parameters have been dropped. The `ezplatform.installer.db_based_installer` service definition has
+  been dropped as well. Instead, the FQCN-named service
+  `EzSystems\PlatformInstallerBundle\Installer\DbBasedInstaller` is available.
+
+* `EzSystems\PlatformInstallerBundle\Command\InstallPlatformCommand` has been marked as final.
+  Overriding it was never supported. Use SchemaBuilder event-oriented extension point to inject
+  custom behavior into the installation process (see eZ Platform Documentation for more details).
```
Comment (alongosz) [inline doc/bc/changes-8.0.md]: @ezsystems/documentation-team if you have time, please review the changes in this doc.

File: doc/bc/changes-8.0.md [hunk 2]
```diff
@@ -55,6 +55,22 @@ Changes affecting version compatibility with former or future versions.
 * Deprecated `eZ\Publish\SPI\FieldType\EventListener` interface, `eZ\Publish\SPI\FieldType\Event` class and
   `eZ\Publish\SPI\FieldType\Events` namespace have been dropped.
 
+* Clean Installer (`EzSystems\PlatformInstallerBundle\Installer\CleanInstaller`), its service
+  definition `ezplatform.installer.clean_installer`, and the
```
Comment (michalmilc90) [inline doc/bc/changes-8.0.md]:
```suggestion
  definition (`ezplatform.installer.clean_installer`), and the
```

File: doc/bc/changes-8.0.md [hunk 3]
```diff
@@ -55,6 +55,22 @@ Changes affecting version compatibility with former or future versions.
...
+  `ezplatform.installer.clean_installer.class` parameter have been dropped. Use Core Installer
+  instead (`EzSystems\PlatformInstallerBundle\Installer\CoreInstaller`) along with
+  `EzSystems\DoctrineSchema\API\Event\SchemaBuilderEvents::BUILD_SCHEMA` event (or just that event).
```
Comment (michalmilc90) [inline doc/bc/changes-8.0.md]:
```suggestion
  the `EzSystems\DoctrineSchema\API\Event\SchemaBuilderEvents::BUILD_SCHEMA` event (or just that event).
```

File: doc/bc/changes-8.0.md [hunk 4]
```diff
@@ -55,6 +55,22 @@ Changes affecting version compatibility with former or future versions.
...
+  See eZ Platform Documentation for more details.
+
+* The `ezplatform.installer.db_based_installer.class` and `ezplatform.installer.install_command.class`
+  parameters have been dropped. The `ezplatform.installer.db_based_installer` service definition has
+  been dropped as well. Instead, the FQCN-named service
+  `EzSystems\PlatformInstallerBundle\Installer\DbBasedInstaller` is available.
+
+* `EzSystems\PlatformInstallerBundle\Command\InstallPlatformCommand` has been marked as final.
```
Comment (michalmilc90) [inline doc/bc/changes-8.0.md]:
```suggestion
* The `EzSystems\PlatformInstallerBundle\Command\InstallPlatformCommand` has been marked as final.
```

File: doc/bc/changes-8.0.md [hunk 5]
```diff
@@ -55,6 +55,22 @@ Changes affecting version compatibility with former or future versions.
...
+  Overriding it was never supported. Use SchemaBuilder event-oriented extension point to inject
```
Comment (michalmilc90) [inline doc/bc/changes-8.0.md]:
```suggestion
  Overriding it was never supported. Use the SchemaBuilder event-oriented extension point to inject
```

File: doc/bc/changes-8.0.md [hunk 6]
```diff
@@ -55,6 +55,22 @@ Changes affecting version compatibility with former or future versions.
 * Deprecated `eZ\Publish\SPI\FieldType\EventListener` interface, `eZ\Publish\SPI\FieldType\Event` class and
   `eZ\Publish\SPI\FieldType\Events` namespace have been dropped.
 
+* Clean Installer (`EzSystems\PlatformInstallerBundle\Installer\CleanInstaller`), its service
+  definition `ezplatform.installer.clean_installer`, and the
+  `ezplatform.installer.clean_installer.class` parameter have been dropped. Use Core Installer
```
Comment (DominikaK) [inline doc/bc/changes-8.0.md]: How about something like:
```suggestion
  `ezplatform.installer.clean_installer.class` parameter have been dropped. Instead use
```

File: doc/bc/changes-8.0.md [hunk 7]
```diff
@@ -55,6 +55,22 @@ Changes affecting version compatibility with former or future versions.
...
+  instead (`EzSystems\PlatformInstallerBundle\Installer\CoreInstaller`) along with
```
Comment (DominikaK) [inline doc/bc/changes-8.0.md]:
```suggestion
  the `EzSystems\DoctrineSchema\API\Event\SchemaBuilderEvents::BUILD_SCHEMA` event
  with an event subscriber, and if needed, additionally with the Core Installer
  (`EzSystems\PlatformInstallerBundle\Installer\CoreInstaller`).
```

File: doc/bc/changes-8.0.md (hunk 3 above)
Comment (DominikaK) [inline doc/bc/changes-8.0.md]:
```suggestion
```

File: doc/bc/changes-8.0.md (hunk 1 above)
Comment (alongosz) [reply doc/bc/changes-8.0.md]: > @ezsystems/documentation-team if you have time, please review the changes in this doc.

Thanks for the review, applied your suggestions.

**Comment by alongosz** [comment]
@michalmilc90 @DominikaK applied your review suggestions via a8bf828.

**Comment by alongosz** [comment]
Dear Reviewers, added:
- 7ec6df356 `[Tests] Autoloaded dev namespace for PlatformInstaller Bundle tests`
- 9f37607f2 `[DX] Added checking if the required Doctrine Schema Bundle is enabled`
- 1551a0960 `[Tests] Added test coverage for DI of InstallerExtension`

File: eZ/Bundle/PlatformInstallerBundle/src/Command/InstallPlatformCommand.php [hunk 8]
```diff
@@ -41,12 +41,12 @@ class InstallPlatformCommand extends Command
...
+        Connection $connection,
         array $installers,
         CacheItemPoolInterface $cachePool,
         $environment
     ) {
-        $this->db = $db;
+        $this->connection = $connection;
         $this->installers = $installers;
         $this->cachePool = $cachePool;
         $this->environment = $environment;
```
Comment (mikadamczyk) [inline eZ/Bundle/PlatformInstallerBundle/src/Command/InstallPlatformCommand.php]: If you touched class property maybe it is worth to add `string` as a type hint for `$this->environment`

File: eZ/Bundle/PlatformInstallerBundle/src/EzSystemsPlatformInstallerBundle.php [hunk 9]
```diff
@@ -8,17 +8,31 @@
...
 use EzSystems\PlatformInstallerBundle\DependencyInjection\Compiler\InstallerTagPass;
-use EzSystems\PlatformInstallerBundle\DependencyInjection\Compiler\SchemaBuilderInstallerPass;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Exception\RuntimeException;
 use Symfony\Component\HttpKernel\Bundle\Bundle;
 
 class EzSystemsPlatformInstallerBundle extends Bundle
 {
+    /**
+     * @param \Symfony\Component\DependencyInjection\ContainerBuilder $container
```
Comment (mikadamczyk) [inline eZ/Bundle/PlatformInstallerBundle/src/EzSystemsPlatformInstallerBundle.php]: not needed

File: doc/bc/changes-8.0.md [hunk 10]
```diff
@@ -63,6 +63,23 @@ Changes affecting version compatibility with former or future versions.
 
 * Deprecated Symfony framework templating component integration has been dropped.
 
+* Clean Installer (`EzSystems\PlatformInstallerBundle\Installer\CleanInstaller`), its service
+  definition (`ezplatform.installer.clean_installer`), and the
```
[the remark this answers is not in the document]
Comment (alongosz) [reply doc/bc/changes-8.0.md]: @mikadamczyk this comma was not by accident there, it's called [Oxford Comma](https://www.lexico.com/en/explore/what-is-the-oxford-comma). Though I might be wrong about the usage in this case, so @michalmilc90 if you could confirm.

[the remark this answers is not in the document]
Comment (mikadamczyk) [reply doc/bc/changes-8.0.md]: good to know 👍

File: eZ/Bundle/PlatformInstallerBundle/src/Command/InstallPlatformCommand.php (hunk 8 above)
Comment (alongosz) [reply eZ/Bundle/PlatformInstallerBundle/src/Command/InstallPlatformCommand.php]: Good catch, fixed.

File: eZ/Bundle/PlatformInstallerBundle/src/EzSystemsPlatformInstallerBundle.php (hunk 9 above)
Comment (alongosz) [reply eZ/Bundle/PlatformInstallerBundle/src/EzSystemsPlatformInstallerBundle.php]: > not needed

Removed.

**Comment by lserwatka** [comment]
@alongosz could you re-base and merge all related PRs?

**Comment by alongosz** [comment]
The entire chain of PRs for [EZP-30898](https://jira.ez.no/browse/EZP-30898) and [EZP-30624](https://jira.ez.no/browse/EZP-30624) have been merged.

**Pull request merged**